geomesa-spark/geomesa-spark-jts/src/test/scala/org/locationtech/geomesa/spark/jts/udf/GeometricCastFunctionsTest.scala

```diff
@@ -31,44 +32,105 @@ class GeometricCastFunctionsTest extends Specification with TestEnvironment {
       "null" >> {
         sc.sql("select st_castToPoint(null)").collect.head(0) must beNull
         dfBlank.select(st_castToPoint(lit(null))).first must beNull
+
+        import spark.implicits._
+        dfBlank.select(st_castToPoint(lit(null)) as 'geom).as[PointContainer].head must haveClass[PointContainer]
       }
 
       "point" >> {
         val pointTxt = "POINT(1 1)"
         val point = s"st_geomFromWKT('$pointTxt')"
-        val df = sc.sql(s"select st_castToPoint($point)")
+        val df = sc.sql(s"select st_castToPoint($point) geom")
         df.collect.head(0).asInstanceOf[AnyRef] must haveClass[Point]
         dfBlank.select(st_castToPoint(st_geomFromWKT(pointTxt))).first must haveClass[Point]
+
+        import spark.implicits._
+        df.as[PointContainer].head must haveClass[PointContainer]
       }
     }
 
     "st_castToPolygon" >> {
       "null" >> {
         sc.sql("select st_castToPolygon(null)").collect.head(0) must beNull
         dfBlank.select(st_castToPolygon(lit(null))).first must beNull
+
+        import spark.implicits._
+        dfBlank.select(st_castToPolygon(lit(null)) as 'geom).as[PolygonContainer].head must haveClass[PolygonContainer]
       }
 
       "polygon" >> {
         val polygonTxt = "POLYGON((1 1, 1 2, 2 2, 2 1, 1 1))"
         val polygon = s"st_geomFromWKT('$polygonTxt')"
-        val df = sc.sql(s"select st_castToPolygon($polygon)")
+        val df = sc.sql(s"select st_castToPolygon($polygon) geom")
```

Comment:
@mojodna, this is maybe a dumb question on my part...  What does adding 'geom' to the end of this SQL statement do?

Reply:
It creates a better alias for the column created. With this, the column is `geom` within the DataFrame and will automatically map to the container's `geom` property. Previously, the column name was `st_castToPolygon(<something>)`.